src/tests/ftest/io/dfuse_find_cmd.py

```diff
@@ -0,0 +1,273 @@
+        """
+        cont_count = self.params.get("cont_count", '/run/container/*')
+        dfs_path = self.params.get("dfs_path", '/run/find_cmd/*')
+        samples = self.params.get("samples", '/run/find_cmd/*')
+        height = self.params.get("height", '/run/find_cmd/*')
+        subdirs_per_node = self.params.get(
+            "subdirs_per_node", '/run/find_cmd/*')
+        files_per_node = self.params.get("files_per_node", '/run/find_cmd/*')
+        needles = self.params.get("needles", '/run/find_cmd/*')
+        challenger_path = self.params.get("challenger_path", '/run/find_cmd/*')
+        temp_dfs_path = ""
+
+        dfuses = list()
+        containers = list()
+
+        self.add_pool(connect=False)
+
+        if not dfs_path:
+            temp_dfs_path = _generate_temp_path_name("/tmp", "dfs_test_")
+            dfs_path = temp_dfs_path
+
+        if challenger_path:
+            challenger_path = _generate_temp_path_name(
+                challenger_path, "test_")
+
+        def _run_test(test_root, dir_tree_paths):
+
+            self._crate_dir_trees(
+                dir_tree_paths,
+                height,
+                subdirs_per_node,
+                files_per_node,
+                needles)
+
+            test_stats = self._run_test(
+                test_root, samples, cont_count, needles)
+
+            return test_stats
+
+        try:
+            mount_dirs = self._setup_containers(
+                dfs_path, cont_count, dfuses, containers)
+
+            daos_stats = _run_test(dfs_path, mount_dirs)
+
+            if challenger_path:
+                challenger_dirs = self._setup_challenger(
+                    challenger_path, cont_count)
+                challenger_stats = _run_test(challenger_path, challenger_dirs)
+
+        except CommandFailure as error:
+            self.log.error("FindCmd Test Failed: %s", str(error))
+            self.fail("Test was expected to pass but it failed.\n")
+        except Exception as error:
+            self.log.error("FindCmd Test Failed: %s", str(error))
+            raise
+        finally:
+            self._teardown_dfuse(dfuses)
+            self.destroy_containers(containers)
+            self.pool.destroy()
+
+            if challenger_path:
+                self._run_cmd(u"rm -rf {0}".format(challenger_path))
+
+            if temp_dfs_path:
+                self._run_cmd(u"rm -rf {0}".format(temp_dfs_path))
+
+        self.log.info("DAOS Stats")
+        daos_stats.print_stats()
+
+        if challenger_path:
+            self.log.info("Challenger Stats")
+            challenger_stats.print_stats()
+
+            self._compare_results(daos_stats, challenger_stats, "all_files")
+            self._compare_results(daos_stats, challenger_stats, "same_suffix")
+            self._compare_results(daos_stats, challenger_stats, "unique_file")
+
+    def _compare_results(self, daos_stats, challenger_stats, tag):
```

Comment:
Sure, I will add descriptions and remove "_".
I am not sure if we need to split this test into two tests.

The original request was to run find(1) on a directory tree created with containers and dfuse. If find(1) crashes, the test failed. If the number of files founded doesn’t match the number of files that were created, the test failed as well. This is what is currently running on this PR.

A second request (commented on DAOS-5563) was to add a performance aspect to compare DAOS performance. So, when the challenger path is provided in the yaml file (ex challenger_path: /mnt/lustre). The test compares the performance. I agree that rather than printing a warning message the test should fail. However, in my experiments DAOS seems to be much slower while running find(1). I am not sure if we have lustre on CI nodes. That's why this feature is not executed in this PR.

I will appreciate feedback with this dilemma.

Reply:
> I thought our convention is not to use _ in private method names. Maybe you can bring it up in a Skype channel or the test meeting.

We do use "_" to indicate "private" class methods.

Reply:
I see. I understand that automating performance test is usually challenging. Thanks for the explanation.